### PR TITLE
Improve interactive AWS creds flow

### PR DESCRIPTION
**Pull request by dschep**
## What did you implement:

closes PLAT-1341

## How did you implement it:
implemented by asking if the user has an account, if not, open browser to sign up
and then open browser (in new or existing acct case) to create an admin user for
user with serverless.

added open as a dependency to open the browser

## How can we verify it:
```
npm i -g serverless/serverless#nicer-aws-creds-flow
mv ~/.aws ~/.awsbak
sls
```

## Todos:

_**Note: Run `npm run test-ci` to run all validation checks on proposed changes**_

- [ ] Write tests and confirm existing functionality is not broken.  
       **Validate via `npm test`**
- [ ] Write documentation
- [ ] Ensure there are no lint errors.  
       **Validate via `npm run lint-updated`**  
       _Note: Some reported issues can be automatically fixed by running `npm run lint:fix`_
- [ ] Ensure introduced changes match Prettier formatting.  
       **Validate via `npm run prettier-check-updated`**  
       _Note: All reported issues can be automatically fixed by running `npm run prettify-updated`_
- [ ] Make sure code coverage hasn't dropped
- [ ] Provide verification config / commands / resources
- [ ] Enable "Allow edits from maintainers" for this PR
- [ ] Update the messages below

**_Is this ready for review?:_** NO  
**_Is it a breaking change?:_** NO

### CODE REVIEW

**Comment by dschep** [comment]
Uggh... `open>=6` depends on node >= 8 but `open<6` fails `npm audit`. Any ideas?

File: lib/utils/openBrowser.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+const chalk = require('chalk');
+const isDockerContainer = require('is-docker');
+const BbPromise = require('bluebird');
+
+function displayManualOpenMessage(url, err) {
+  // https://github.com/sindresorhus/log-symbols
+  console.log('---------------------------');
+  const errMsg = err ? `\nError: ${err.message}` : '';
+  const msg = `Unable to open browser automatically${errMsg}`;
+  console.log(`🙈  ${chalk.red(msg)}`);
```
Comment (pmuens) [inline lib/utils/openBrowser.js]: While cute I'm unsure if we want to go the "emotification of all the CLIs" route in an enterprise-facing project.

Comment (dschep) [reply lib/utils/openBrowser.js]: I copy and pasted this out of `platform-sdk` 😬

Comment (dschep) [reply lib/utils/openBrowser.js]: (as to why i copied, it.. didn't want to add sdk as a direct dep, which is the only way to guarantee it's availability with things like `yarn`, and now also the fact it requires newer versions of node)

**Comment by dschep** [comment]
> 🤔 Maybe we can to fork open it and fix the dependency issue for version 6 on our own?

Yeah.. It's a single file AFAICT https://github.com/sindresorhus/open/blob/master/index.js So i'll do that.

File: lib/utils/openBrowser.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+/* eslint-disable no-console */
+
+const opn = require('./open');
+const chalk = require('chalk');
+const isDockerContainer = require('is-docker');
+const BbPromise = require('bluebird');
+
+function displayManualOpenMessage(url, err) {
+  // https://github.com/sindresorhus/log-symbols
+  console.log('---------------------------');
```
Comment (medikoo) [inline lib/utils/openBrowser.js]: Why we do not rely here on serverless logger? Even if we don't like to use `cli.log` we may rel on `cli.consoleLog`.

It'll clearly indicate the intention of generating user facing CLI output.

Direct use of `console.log` is more for temporary debug output (which we prevent from committing in via lint rules)

File: lib/utils/openBrowser.js (hunk 1 above)
Comment (medikoo) [reply lib/utils/openBrowser.js]: > now also the fact it requires newer versions of node

But SDK Is dependency of a plugin.. so if SDK was updated to not work in v6, then through Plugin  it'll also break SLS (?)

**Pull request merged**